tests/src/test/java/com/maxio/advancedbilling/controllers/subscriptions/SubscriptionsControllerPurgeTest.java

```diff
@@ -0,0 +1,153 @@
+        assertNotFound(() -> SUBSCRIPTIONS_CONTROLLER
+                .purgeSubscription(123, customer.getId(), emptyList()));
+    }
+
+    @ParameterizedTest
+    @MethodSource("argsForShouldPurgeSubscriptionWithCustomerAndPaymentProfile")
+    void shouldPurgeSubscriptionWithCustomerAndPaymentProfile(List<SubscriptionPurgeType> cascade) throws IOException, ApiException {
+        // given
+        Subscription subscription = TEST_SETUP.createSubscription(customer, product);
+
+        // when
+        SUBSCRIPTIONS_CONTROLLER.purgeSubscription(
+                subscription.getId(),
+                customer.getId(),
+                cascade
+        );
+
+        // then
+        assertNotFound(() -> SUBSCRIPTIONS_CONTROLLER.readSubscription(subscription.getId(), emptyList()));
+        assertNotFound(() -> CUSTOMERS_CONTROLLER.readCustomer(customer.getId()));
+        assertThat(PAYMENT_PROFILES_CONTROLLER.listPaymentProfiles(new ListPaymentProfilesInput.Builder().customerId(customer.getId()).build()))
+                .isEmpty();
+    }
+
+    private static Stream<Arguments> argsForShouldPurgeSubscriptionWithCustomerAndPaymentProfile() {
+        return Stream.of(
+                arguments(List.of(SubscriptionPurgeType.CUSTOMER, SubscriptionPurgeType.PAYMENT_PROFILE)),
+                arguments(List.of(SubscriptionPurgeType.CUSTOMER))
+        );
+    }
+
+    @Test
+    void shouldPurgeSubscriptionWithPaymentProfileLeavingCustomer() throws IOException, ApiException {
+        // given
+        Subscription subscription = TEST_SETUP.createSubscription(customer, product);
+
+        // when
+        SUBSCRIPTIONS_CONTROLLER.purgeSubscription(
+                subscription.getId(),
+                customer.getId(),
+                List.of(SubscriptionPurgeType.PAYMENT_PROFILE)
+        );
+
+        // then
+        assertNotFound(() -> SUBSCRIPTIONS_CONTROLLER.readSubscription(subscription.getId(), emptyList()));
+        assertThat(CUSTOMERS_CONTROLLER.readCustomer(customer.getId()).getCustomer()).isNotNull();
+        assertThat(PAYMENT_PROFILES_CONTROLLER.listPaymentProfiles(new ListPaymentProfilesInput.Builder().customerId(customer.getId()).build()))
+                .isEmpty();
+    }
+
+    @Test
+    void shouldReturn400WhenAckCustomerIsNotSubscriptionOwner() throws IOException, ApiException {
+        // given
+        Subscription subscription = TEST_SETUP.createSubscription(customer, product);
+
+        // when - then
+        new ApiExceptionAssert(() -> SUBSCRIPTIONS_CONTROLLER
+                .purgeSubscription(
+                        subscription.getId(),
+                        customerWithoutSubscription.getId(),  // not the subscription owner
+                        emptyList()
+                )
+        )
+                .isBadRequest()
+                .hasMessageStartingWithHttpNotOk();
+        // not asserting the whole message because it contains the entire subscription's string representation
```

Comment:
Can you get Subscription model from the response? Add matcher like for other specific error responses to assert on the error model.